docs/DeveloperGuide.adoc

```diff
@@ -244,6 +243,32 @@ _{Explain here how the data encryption feature will be implemented}_
 
 // end::dataencryption[]
 
+// tag::reminder[]
+=== Reminder Feature
+==== Implementation
+A bookmark can have a reminder, and a reminder can use to open the bookmark.
+This mechanism is facilitated by `Reminder` and `ReminderAssociation`.
+
+`Reminder` containing a reminding time, the `Url` of the `Bookmark` and a `Note`.
+
+`ReminderAssociation` represents the association between `Reminder` and `Bookmark`, containing
+ the relation from `Reminder` to `Bookmark` and `Reminder` to `Bookmark`.
+
+.Structure of the Reminder
+image::ReminderAssociationDiagram.png[]
+
+==== Design Consideration
+===== Aspect: How to represent the relationship between reminder and bookmark.
+* **Alternative 1 (current choice):** Use two hashmap to store both the relationship from reminder to bookmark and bookmark to reminder.
+** Pros: More OOP and can easily find the relationship.
+** Cons: More complex when doing any operation to a reminder. Need to check two hashmap.
+
+* **Alternative 2:** Reminder is a field of bookmark.
+** Pros: Easier to implement.
+** Cons: Reminder is a field of bookmark, so using a reminder to open the bookmark violate the relationship.
```

Comment:
violates what relationship ah?

Reply:
It like Bookmark --> Reminder, but now using reminder to open bookmark will be Reminder --> Bookmark. I don't how to describe it precisely.

Reply:
```suggestion
//TODO: Change this description
** Cons: Reminder is a field of bookmark, so using a reminder to open the bookmark violate the relationship.
```